moveit_ros/planning_interface/move_group_interface/src/wrap_python_move_group.cpp

```diff
@@ -513,6 +513,27 @@ class MoveGroupInterfaceWrapper : protected py_bindings_tools::ROScppInitializer
       return "";
     }
   }
+
+  bp::list getJacobianMatrixPython(bp::list& joint_values)
+  {
+    std::vector<double> v = py_bindings_tools::doubleFromList(joint_values);
+    robot_state::RobotState state(getRobotModel());
+    state.setToDefaultValues();
+    auto group = state.getJointModelGroup(getName());
+    state.setJointGroupPositions(group, v);
+    Eigen::MatrixXd matrix = state.getJacobian(group);
```

Comment:
Wouldn't it be possible to return the Jacobian as a `numpy` matrix?
In this case, there will be only one (costly) memory allocation instead of `rows+1`.

Reply:
Thank you, I revised it  to return a numpy matrix. Firstly I thought to use boost.numpy, but libboost seems not support numpy module in Ubuntu 16.04, so I used eigenpy. The returned value is numpy.matrixlib.defmatrix.matrix (not numpy.ndarray), but I believe it is not a problem.